### PR TITLE
Adding shelter territory and testing it. also testing report method

### PR DESCRIPTION
**Pull request by Kamen06**

### CODE REVIEW

File: src/animalShelter/AnimalShelter.ts [hunk 1]
```diff
@@ -44,6 +47,14 @@ export default class AnimalShelter {
     this.animals = [];
     this._mammalLimit = DEFAULT_MAMMAL_LIMIT;
     this._reptileLimit = DEFAULT_REPTILE_LIMIT;
+    this._shelterTerritory = DEFAULT_SHELTER_TERRITORY;
+  }
+  getShelterTerritory() {
```
Comment (rkoychev) [inline src/animalShelter/AnimalShelter.ts:52]: no return type specified

File: src/animalShelter/AnimalShelter.ts [hunk 2]
```diff
@@ -85,24 +96,28 @@ export default class AnimalShelter {
...
+
+    } = {
+      animalsWhoJump: 0,
+      animalsWhoRun: 0,
+      anaimalsWhoClimbTrees: 0
+    }
     for (const animal of this.animals) {
       if (animal.canJump) {
-        animalsWhoJump++
+        animalNumbers.animalsWhoJump++
```
Comment (rkoychev) [inline src/animalShelter/AnimalShelter.ts:111]: missing semicolon

File: src/animalShelter/AnimalShelter.ts [hunk 3]
```diff
@@ -85,24 +96,28 @@ export default class AnimalShelter {
...
 
   public report() {
-    let animalsWhoJump = 0;
-    let animalsWhoRun = 0;
-    let anaimalsWhoClimbTrees = 0
+    const animalNumbers: {
+      animalsWhoJump: number,
+      animalsWhoRun: number,
+      anaimalsWhoClimbTrees: number
+
```
Comment (rkoychev) [inline src/animalShelter/AnimalShelter.ts:103]: additional empty row

File: tests/animalShelter.test.ts [hunk 4]
```diff
@@ -115,4 +115,24 @@ describe("Animal Shelter Tests", () => {
     expect(animals.mamamls).toBe(3);
     expect(animals.reptiles).toBe(2);
   });
+  test("should succesfully set shelter trritory", () => {
+    const animalShelter = AnimalShelter.getInstance();
+    animalShelter.setShelterTerritory(2000);
```
Comment (rkoychev) [inline tests/animalShelter.test.ts:120]: extract 2000 in constant

Comment (rkoychev) [reply tests/animalShelter.test.ts:120]: do you happen to test for a territory of 0 or negative ?

**Review by rkoychev — CHANGES_REQUESTED** [review]

File: tests/animalShelter.test.ts (hunk 4 above)
Comment (Kamen06) [reply tests/animalShelter.test.ts:120]: why do i need to extract a number in the tests? Following this logic i need to extract every number in the test folder

**Review by rkoychev — APPROVED** [review]
You need to merge main branch and use the prettier settings before you can merge this.

**Pull request merged**